includes/wikia/api/RecommendationTileApiController.class.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/**
+ * Retrieves the content of tiles displayed by the recommendations module.
+ * This controller supports a single wiki. If data regarding articles from multiple wikis is required, you should make multiple call with a modified wiki id header.
+ */
+class RecommendationTileApiController extends WikiaApiController {
+
+	const THUMBNAIL_MIN_SIZE = 400;
+	const CACHE_3_DAYS = 259200;
+
+	function getDetails() {
+		$this->setOutputFieldType( "items", self::OUTPUT_FIELD_TYPE_OBJECT );
+		$articleIds = str_getcsv( $this->request->getVal( 'ids', null ) );
+
+		$wikiId = $this->getWikiId();
+		$wikiName = $this->getWikiName();
+
+		$articles = $this->getArticleProperties( $articleIds );
+		$thumbnails = $this->getArticlesThumbnails( $articleIds, self::THUMBNAIL_MIN_SIZE, self::THUMBNAIL_MIN_SIZE );
+		$videoIds = $this->getFeaturedVideos();
+
+		$items = [];
+		/** @var Title $articleTitle */
+		foreach ( $articles as $articleTitle ) {
+			$articleId = $articleTitle->getArticleID();
+			$thumbnail = $thumbnails[$articleId];
+			$url = $articleTitle->getFullURL();
+
+			$items[$wikiId . '_' . $articleId] = [
+				// IW-2588: Force these URLs to always use HTTPS
+				// The recommendations service calls this API over HTTP, but forwards these URLs to clients
+				'url' => wfHttpsAllowedForURL( $url ) ? wfHttpToHttps( $url ) : $url,
+				'title' => $articleTitle->getPrefixedText(),
+				'wikiName' => $wikiName,
+				'thumbnail' => $thumbnail,
+				'hasVideo' => in_array( $articleId, $videoIds ),
+			];
+		}
+
+		$this->setResponseData(
+			[ 'items' => $items ],
+			null,
+			# cache for 3 days to avoid daily cache hit ratio drops - https://wikia-inc.atlassian.net/browse/DE-4346
+			static::CACHE_3_DAYS
+		);
+	}
+
+	private function getArticlesThumbnails( $articleIds, int $width, int $height ): iterable {
+		$ids = !is_array( $articleIds ) ? [ $articleIds ] : $articleIds;
```

Comment:
$articleIds should always be an array here, so we could remove this check and add a type hint